### PR TITLE
Added platformIO

### PR DESCRIPTION
**Pull request by Samureimer**
Added a config file for PlatformIO as I felt it was missing from the project.
Also updated some library versions and removed the RingBuffer library as it seems like it is no longer used.

All build flags comes from this issue:
https://github.com/Egyras/HeishaMon/issues/277

### CODE REVIEW

**Comment by Samureimer** [comment]
@Egyras or any other maintainer, could you have a look at this PR?

**Comment by geduxas** [comment]
> @Egyras or any other maintainer, could you have a look at this PR?

I don't think it's right config for build first of all default (original build) board is d1mini and cpu is clocked to 160Mhz.

**Comment by Samureimer** [comment]
I just use a NodeMcu and forgot to add the D1 Mini.
Also, I couldn't find any references to the board running 160Mhz but I added it in the config.

**Comment by geduxas** [comment]
> I just use a NodeMcu and forgot to add the D1 Mini. Also, I couldn't find any references to the board running 160Mhz but I added it in the config.

All build rules can be found in .workflow directory there is rules for github to build from source. 

`` arduino-cli compile --output-dir . --fqbn=esp8266:esp8266:d1_mini:xtal=160,vt=flash,ssl=basic,mmu=3216,non32xfer=safe,eesz=4M2M,ip=lm2f,dbg=Disabled,lvl=None____,wipe=none,baud=921600 --vid-pid=1A86_7523 --warnings=none --verbose HeishaMon.ino
``

File: platformio.ini [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	knolleary/PubSubClient@^2.8
+	links2004/WebSockets@^2.4.0
+	bblanchon/ArduinoJson@^6.19.4
+	milesburton/DallasTemperature@^3.9.0
+	jenscski/DoubleResetDetect@^1.0.0
+build_flags =
+	-DMMU_IRAM_SIZE=0x8000
+	-DMMU_ICACHE_SIZE=0x4000
+	-DMMU_SEC_HEAP=0x40108000
+	-DMMU_SEC_HEAP_SIZE=0x4000
```
Comment (geduxas) [inline platformio.ini]: Also, acording [platformio](https://docs.platformio.org/en/stable/platforms/espressif8266.html#mmu-adjusting-icache-to-iram-ratio) documentation, this should be enought
```
build_flags =
	-D PIO_FRAMEWORK_ARDUINO_MMU_CACHE16_IRAM32_SECHEAP_NOTSHARED
```

**Pull request merged**